### DISCUSSION TITLE
[Enhancement] Notification icon with context menu

### DISCUSSION THREAD

**Discussion by Lakritzator** [dotnet/maui#130] [General]
## Summary
Provide a multi platform component to support the concept of a notification icon, the main interaction point for a background service / application. This is what Linux, MacOs and Windows all provide, even Android has something similar.

## API Changes
Provide a way to specify an icon, show it in the notification area/center, hide, or change the icon. Have ways to handle mouse & keyboard input, and show UI elements like a context menu or some other notification. 

## Intended Use Case
When an application currently doesn't have any user interaction, e.g. it's waiting for a hotkey or some even to happen, it might still want to provide a way for the user to interact with it or just see that it's running. A solution would be to show an icon in the notification area, which the user can click. On a click the application might want to show a context menu, configuration, or a window.

An example of this functionality is the NotifyIcon available with Windows Forms, or the open source [WPF Notify icon](https://github.com/hardcodet/wpf-notifyicon) project, which I contribute to. I'm not aware of the names in other operating systems...

**Comment by Lakritzator** [comment]
This might be complemented with support for notifications.

**Comment by mrwcjoughin** [comment]
all of what you're talking about preceded push notifications. 
To me, this can be handled by push notifications and that's all that's needed.
I would imagine the Xamarin/MAUI team will feel the same...

**Comment by Lakritzator** [comment]
I feel like either you didn't read all of it, or I described or I described it horribly.

I think push notifications are a completely different use-case, I'm talking about desktop applications without a central service. And I'm taking about an icon to control the application, push notifications don't help.

**Comment by dishantmshah** [comment]
Notification Icons is not yet supported by UWP framework as well. It only works for WPF/Win32 applications, also I don't see any use case to implement it for Android/iOS as they already have Push Notifications, which can be used with background services as a replacement for notification icons.

**Comment by Lakritzator** [comment]
Push messages are not the same as an icon which is an interaction point for the user to open a context menu or other windows (maybe close the app). Having a platform agnostic way for notifications is just a nice to have addition, let's strike it from this request.

This request makes sense for MacOS, Windows and Linux.
